contracts/test/ClaimTest.t.sol

```diff
@@ -578,4 +621,281 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
+
+        assertEq(details[0], principalPortion + interestPortion);
+        assertEq(details[1], interestPortion);
+        assertEq(details[2], principalPortion);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 0);
+
+        _assertPoolState(principalPortion, interestPortion);
+
+        pool.triggerDefault(address(loanV2), address(debtLockerFactory));
+
+        try pool.claim(address(loanV2), address(debtLockerFactory)) { 
+            assertTrue(false, "Claim before liquidation is done"); 
+        } catch Error(string memory reason) {
+            assertEq(reason, "DL:HCOR:LIQ_NOT_FINISHED");
+        }
+
+        _liquidateCollateral();
+
+        details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], 280_135_620000);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 280_135_620000);
+        assertEq(details[6], 719_864_380000);
+    }
+
+    function test_claim_liquidationDOS() external {
+        _createLoan(INTEREST_ONLY, UNDER_COLLATERALIZED);
+        _fundLoan();
+        _drawdownLoan();
+
+         // Put in default
+        hevm.warp(block.timestamp + loanV2.nextPaymentDueDate() + loanV2.gracePeriod() + 1);
+
+        pool.triggerDefault(address(loanV2), address(debtLockerFactory));
+
+        try pool.claim(address(loanV2), address(debtLockerFactory)) { 
+            assertTrue(false, "Claim before liquidation is done"); 
+        } catch Error(string memory reason) {
+            assertEq(reason, "DL:HCOR:LIQ_NOT_FINISHED");
+        }
+
+        _liquidateCollateral();
+
+        // Malicious entity will send 1 WEI work of collateral to Liquidator
+        DebtLocker debtLocker =  DebtLocker(pool.debtLockers(address(loanV2), address(debtLockerFactory)));
+
+        address liquidator = debtLocker.liquidator();
+        
+        erc20_mint(WBTC, 0, liquidator, 1);
+
+        // Claiming is locked again
+        try pool.claim(address(loanV2), address(debtLockerFactory)) { 
+            assertTrue(false, "Claim before liquidation is done"); 
+        } catch Error(string memory reason) {
+            assertEq(reason, "DL:HCOR:LIQ_NOT_FINISHED");
+        }
+
+        debtLocker.pullFundsFromLiquidator(address(liquidator), WBTC, address(this), 1);
+
+        uint256[7] memory details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], 280_135_620000);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 280_135_620000);
+        assertEq(details[6], 719_864_380000);
+    }
+
+    function test_claim_liquidatonStop() external {
```

Comment:
test_claim_liquidatonStop_before_liquidation

Reply:
Should add a comment outlining that this is a very bad thing, and add some code to pull funds and show how the collateralAsset can be manually recovered after this